feature/home/src/main/java/com/epmedu/animeal/home/presentation/viewmodel/HomeViewModel.kt

```diff
@@ -46,19 +57,27 @@ class HomeViewModel @Inject constructor(
     fun handleEvents(event: HomeScreenEvent) = when (event) {
         is HomeScreenEvent.FeedingPointSelected -> selectFeedingPoint(event)
         is HomeScreenEvent.FeedingPointFavouriteChange -> changeFavouriteFeedingPoint(event)
-        is HomeScreenEvent.ShowWillFeedDialog -> showWillFeedDialog()
-        is HomeScreenEvent.DismissWillFeedDialog -> dismissWillFeedDialog()
-        is HomeScreenEvent.GeolocationPermissionStatusChanged -> changeGeolocationPermissionStatus(event)
+        is HomeScreenEvent.RouteEvent -> {
+            handleRouteEvent(event)
+            when (event) {
+                HomeScreenEvent.RouteEvent.FeedingRouteCancellationRequest -> fetchFeedingPoints()
+                HomeScreenEvent.RouteEvent.FeedingRouteStartRequest -> hideFeedingPointsAndSaveFeeder()
```

Comment:
Why don't you handle both route events in the `RouteHandler`? I see that `fetchFeedingPoints` is used in the viewmodel twice, but you can create one more delegate, move this function in and use it in the viewmodel and route handler